tools/link.py

```diff
@@ -1240,7 +1240,7 @@ def phase_linker_setup(options, state):  # noqa: C901, PLR0912, PLR0915
       state.append_link_flag('--no-whole-archive')
     settings.FILESYSTEM = 1
     settings.SYSCALLS_REQUIRE_FILESYSTEM = 0
-    settings.JS_LIBRARIES.append((0, 'library_wasmfs.js'))
```

Comment:
Did this 0 not ensure that system libs were before user ones?

Reply:
Indeed, and that is still that case.   All these `JS_LIBRARIES.append` happen during `process_libraries` which runs before `process_libraries`.

Reply:
First `process libraries` should be `phase_linker_setup` in the comment? If so then yes, that looks right, thanks.

Reply:
Yes sorry, fixed commend